### PR TITLE
nautilus: fixed mime types

### LINKED ISSUE #12280: gnome3.nautilus does not recognize image mimetypes
@lethalman 
I just installed gnome3.nautilus because apparently kde5.dolphin is broken on nixos.
Sadly my image preview is also broken in nautilus. It looks like this:

screenshot from 2016-01-10 03-25-16

The .jpg file is wrongly recognized as a text/plain file and therefore no preview is generated.

### PR DESCRIPTION
**Pull request by mimame**
Fixed #12280

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/desktops/gnome-3/core/nautilus/default.nix [hunk 1]
```diff
@@ -21,6 +21,7 @@ stdenv.mkDerivation rec {
       # Thumbnailers
       --prefix XDG_DATA_DIRS : "${gdk_pixbuf}/share"
       --prefix XDG_DATA_DIRS : "${librsvg}/share"
+      --prefix XDG_DATA_DIRS : "${shared_mime_info}/share"
```
Comment (Mic92) [inline pkgs/desktops/gnome-3/core/nautilus/default.nix]: How about?

```
       --prefix XDG_DATA_DIRS : "${gdk_pixbuf}/share:${librsvg}/share:${shared_mime_info}/share"
```

Comment (jtojnar) [reply pkgs/desktops/gnome-3/core/nautilus/default.nix]: Or even `--prefix XDG_DATA_DIRS : "${stdenv.lib.makeSearchPath "share" [ gdk_pixbuf librsvg shared_mime_info ]}"`

Comment (mimadrid) [reply pkgs/desktops/gnome-3/core/nautilus/default.nix]: Thanks a lot for the tips. Changed :+1:

**Comment by Mic92** [comment]
@GrahamcOfBorg build gnome3.nautilus

**Pull request merged**